blockchain/state_manager/src/lib.rs

```diff
@@ -98,8 +98,10 @@ where
     pub async fn new(
         cs: Arc<ChainStore<DB>>,
         chain_config: Arc<ChainConfig>,
-    ) -> Result<Self, Box<dyn std::error::Error>> {
-        let genesis = cs.genesis()?.ok_or("genesis header was none")?;
+    ) -> Result<Self, anyhow::Error> {
+        let genesis = cs
+            .genesis()?
+            .ok_or_else(|| anyhow::anyhow!("genesis header was none"))?;
```

Comment:
perhaps `geneis header missing` would be more helpful for those not versed in Rust scriptures